src/utils/blockchain/profile.ts

```diff
@@ -16,11 +17,14 @@ export const depositFreeBond = async (web3Interface: Web3Interface, amt: BigNumb
   setInterval(() => process.stdout.write("."), 1000);
 
   try {
-    const tx = await retryFn(() => web3Interface.archaeologistFacet.depositFreeBond(amt));
+    const tx = await retryFn(async () => {
+      await web3Interface.archaeologistFacet.runMethod("depositFreeBond", amt);
```

Comment:
Demo-ing how `runMethod` might be used. Not sure how I feel about this vs having to basically rewrite contract function signatures in order to use them